datashader/core.py

```diff
@@ -404,7 +424,7 @@ def raster(self,
                       us_method=interpolate, fill_value=fill_value)
         if array.ndim == 2:
             source_window = array[rmin:rmax+1, cmin:cmax+1]
-            if isinstance(source_window, Array):
+            if has_dask and isinstance(source_window, Array):
```

Comment:
Seems like this change could be avoided by defining Array and DaskDataFrame as new empty/stub classes if Dask isn't available; that way there would be no need to check for it here or below.

Reply:
(And similarly for PySpark dataframes).

Reply:
I'm reverting this change for `dask`. I have implemented the suggested change for `pyspark`.